tests/test_ga4_conversion_window_invalid.py

```diff
@@ -0,0 +1,147 @@
+            'oauth_client_id': os.getenv('TAP_GA4_CLIENT_ID'),
+            'user_id': os.getenv('TAP_GA4_USER_ID'),
+            'report_definitions': [
+                {"id": self.custom_report_id_1, "name": "Test Report 1"},
+                {"id": self.custom_report_id_2, "name": "Test Report 2"},
+            ]
+        }
+        if original:
+            return return_value
+
+        if self.start_date:
+            return_value["start_date"] = self.start_date
+        if self.request_window_size:
+            return_value["request_window_size"] = self.request_window_size
+        return return_value
+
+
+    def streams_to_test(self):
+        # testing all streams creates massive quota issues
+        custom_id = self.custom_reports_names_to_ids()['Test Report 1']
+        return {
+            'content_group_report',
+            custom_id
+        }
+
+
+    def run_test(self):
+        """
+        Testing that basic sync functions without Critical Errors when
+        a valid conversion_window is set.
+        """
+
+        try:
+            # Create a connection
+            conn_id = connections.ensure_connection(self, original_properties=False)
+
+            # Run a discovery job
+            found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+            # Select tables and fields
+            test_catalogs = [catalog
+                             for catalog in found_catalogs
+                             if catalog['stream_name'] in self.streams_to_test()]
+            streams_to_selected_fields = {stream: set() for stream in self.streams_to_test()}
+            self.select_streams_and_fields(conn_id, test_catalogs, streams_to_selected_fields)
+
+            # set state to ensure conversion window is used
+            today_datetime = dt.strftime(dt.utcnow(), self.BOOKMARK_FORMAT)
+            initial_state = {
+                'bookmarks': {
+                    stream: { os.getenv('TAP_GA4_PROPERTY_ID'): {'last_report_date': today_datetime}}
+                    for stream in self.streams_to_test()
+                }, 'currently_syncing': None
+            }
+
+            menagerie.set_state(conn_id, initial_state)
+
+            # Run a sync
+            sync_job_name = runner.run_sync_mode(self, conn_id)
+
+            # Verify the tap and target do not throw a critical error
+            exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+            menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+            # Verify tap replicates through today by check state
+            final_state = menagerie.get_state(conn_id)
+            self.assertDictEqual(final_state, initial_state)
+
+            with self.subTest():
+                raise AssertionError("Tap should not have ran sync with conversion_window: "
+                                     f"value {self.conversion_window}, type {type(self.conversion_window)}")
+
+        except Exception as ex:
```

Comment:
Also, this is so broad of an exception it will catch everything including your raise AssertionError above.  We should be more specific if we can when using an except. Although we should remove this and use the approach in the comment abovce

Reply:
I'm not sure what the effect of catching the AssertionError above would be.  In failed test runs were the AssertionError is raised I did not see the except code block doing anything.  Let's discuss.